spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -1,69 +1,108 @@
+  end
+
+  after do
+    # SiteConfig.clear_cache should work here but for some reason it isn't
+    SiteConfig.staff_user_id = 1
+  end
+
   describe "::call" do
-    let(:receiving_user) { create(:user) }
+    it "send all forms of welcome notifications" do
+      user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago)
+      sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+      expect(user.notifications.count).to eq(2)
+    end
+  end
+
+  describe "#send_welcome_notification" do
     let(:user) { create(:user) }
-    let(:mascot_account) { create(:user) }
-    let!(:welcome_broadcast) { create(:welcome_broadcast, :active) }
 
-    before do
-      allow(User).to receive(:mascot_account).and_return(mascot_account)
-      SiteConfig.staff_user_id = mascot_account.id
+    it "generates the correct broadcast type and sends the notification to the user" do
+      sidekiq_perform_enqueued_jobs { described_class.new(user.id).send_welcome_notification }
+      expect(user.notifications.first.notifiable).to eq(welcome_broadcast)
     end
 
-    after do
-      SiteConfig.staff_user_id = 1
+    it "does not send to a user who has commented in a welcome thread" do
+      create(:comment, commentable: welcome_thread, commentable_type: "Article", user: user)
+      expect do
+        sidekiq_perform_enqueued_jobs { described_class.new(user.id).send_welcome_notification }
+      end.not_to change(user.notifications, :count)
     end
 
-    context "when sending a set_up_profile notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has set up their profile"
+    it "does not send a duplicate notifications" do
+      2.times do
+        sidekiq_perform_enqueued_jobs { described_class.new(user.id).send_welcome_notification }
+      end
+
+      expect(user.notifications.count).to eq(1)
     end
+  end
 
-    context "when sending a welcome_thread notification" do
-      before do
-        welcome_thread_article = create(:article, title: "Welcome Thread - v0", published: true, tags: "welcome", user: mascot_account)
-        create(:comment, commentable: welcome_thread_article, commentable_type: "Article", user: user)
-      end
+  context "when sending an authentication notification" do
+    before do
+      allow(Notification).to receive(:send_welcome_notification)
+    end
 
-      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
-        expect(receiving_user.notifications.count).to eq(0)
-        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+    it "does not send notification if user is created less than a day ago" do
+      user = create(:user, :with_identity, identities: ["github"])
```

Comment:
should we extract this github-authed user into a `let`?

Reply:
I prefer to keep it local right now as each spec has a different auth and create_at requirement. For these specs, I get the context faster when keeping variables within the block.